src/Components/WebAssembly/WebAssembly/test/Hosting/TestWebAssemblyJSRuntime.cs

```diff
@@ -1,25 +1,31 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using Microsoft.JSInterop.WebAssembly;
-using Moq;
+using System;
+using Microsoft.AspNetCore.Components.WebAssembly.Services;
 
 namespace Microsoft.AspNetCore.Components.WebAssembly.Hosting
 {
-    public class TestWebAssemblyJSRuntime
+    internal class TestWebAssemblyJSRuntimeInvoker : WebAssemblyJSRuntimeInvoker
     {
-        public static WebAssemblyJSRuntime Create(string environment = "Production")
-        {
-            var jsRuntime = new Mock<WebAssemblyJSRuntime>();
-            jsRuntime.Setup(j => j.InvokeUnmarshalled<object, object, object, string>("Blazor._internal.getApplicationEnvironment", null, null, null))
-                .Returns(environment)
-                .Verifiable();
+        private readonly string _environment;
 
-            jsRuntime.Setup(j => j.InvokeUnmarshalled<string, object, object, byte[]>("Blazor._internal.getConfig", It.IsAny<string>(), null, null))
-                .Returns((byte[])null)
-                .Verifiable();
+        public TestWebAssemblyJSRuntimeInvoker(string environment = "Production")
+        {
+            _environment = environment;
+        }
 
-            return jsRuntime.Object;
+        public override TResult InvokeUnmarshalled<T0, T1, T2, TResult>(string identifier, T0 arg0, T1 arg1, T2 arg2)
+        {
+            switch (identifier)
+            {
+                case "Blazor._internal.getApplicationEnvironment":
```

Comment:
Nit: Does it make sense to attach these values to constants? Feel free to consider this out of scope since it looks like the code was using string literals prior.

Reply:
If any other cases emerge where we reuse the values, we can make consts then. For as long as there’s only one usage I suspect a literal will be fine.